src/services/api/restBuilder/fetcher/fetcher.ts

```diff
@@ -33,9 +31,10 @@ export default async function fetcher(
     );
   }
 
-  // 204 responses do not include content, calling response.json() throws an
-  // exception.
-  if (response.status === 204) return {};
+  if (!response.headers.get("content-length")) {
+    // @ts-expect-error: cannot call .json when there's an empty body.
+    return undefined;
+  }
```

Comment:
I'm not sure what this is for, on my end I don't get the content-length header so this causes the return to always be undefined

Reply:
oof. The `delete` calls from the backend don't have a body, so the `response.json()` fails. Other options:
1. use `response.text()` then parse it ourselves if there's a body
2. try/catch response.json and return undefined if it fails.
In the current and option 1, the function will still throw if the body exist but isn't json, in the second we lose that and just get undefined instead.

Reply:
what was wrong with the previous solution? `if (response.status === 204) return {};` we get a 204 from delete calls and then return an empty object

Reply:
I figure it's to make it more general, not only 204 could have an empty body.. So I went with option 2, which I pushed in my last commit